java-checks/src/main/java/org/sonar/java/checks/TooManyStatementsPerLine_S00122_Check.java

```diff
@@ -47,91 +74,231 @@
 @ActivatedByDefault
 @SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.READABILITY)
 @SqaleConstantRemediation("1min")
-public class TooManyStatementsPerLine_S00122_Check extends SquidCheck<LexerlessGrammar> implements JavaCheck {
-
-  private static final AstNodeType[] STATEMENTS_KINDS = new ImmutableList.Builder<AstNodeType>()
-    .add(
-      Kind.EMPTY_STATEMENT,
-      Kind.LABELED_STATEMENT,
-      Kind.IF_STATEMENT,
-      Kind.ASSERT_STATEMENT,
-      Kind.SWITCH_STATEMENT,
-      Kind.WHILE_STATEMENT,
-      Kind.DO_STATEMENT,
-      Kind.BREAK_STATEMENT,
-      Kind.CONTINUE_STATEMENT,
-      Kind.RETURN_STATEMENT,
-      Kind.THROW_STATEMENT,
-      Kind.SYNCHRONIZED_STATEMENT,
-      Kind.EXPRESSION_STATEMENT,
-      Kind.FOR_STATEMENT,
-      Kind.FOR_EACH_STATEMENT,
-      Kind.TRY_STATEMENT)
-    .build()
-    .toArray(new AstNodeType[0]);
-
-  private final Multiset<Integer> statementsPerLine = HashMultiset.create();
-  private final Map<Integer, Integer> columnsByLine = new HashMap<Integer, Integer>();
+public class TooManyStatementsPerLine_S00122_Check extends SubscriptionBaseVisitor {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/henryju/image-hosting/master/major.png)Rename this class name to match the regular expression '^[A-Z][a-zA-Z0-9]*$'.[![...](https://raw.githubusercontent.com/henryju/image-hosting/master/more.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00101)